doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1728,16 +1736,62 @@ String notations
      concrete string expressed as a decimal.  They may not return
      opaque constants.
 
-The following errors apply to both string and numeral notations:
+.. note::
+   Number or string notations for parameterized inductive types can be
+   added by declaring an :ref:`abbreviation <Abbreviations>` for the
+   inductive which instantiates all parameters. See example below.
+
+The :n:`via` option applies to both string and number notations:
+
+:n:`via @qualid__ind using ( {+, @qualid__constant @qualid__constructor } )`
+   When using this variant, :n:`@qualid__type` no
+   longer needs to be an inductive. It is mapped to the
+   inductive :n:`@qualid__ind` according to the provided
+   list of pairs whose first component :n:`@qualid__constant` is a
+   constant of type :n:`@qualid__type`
+   (or a function of type :n:`{* _ -> } @qualid__type`) and the second a
+   constructor of type :n:`@qualid__ind`. The type
+   :n:`@qualid__type` is then replaced by :n:`@qualid__ind` in the
+   above parser and printer types.
+
+   When :n:`@qualid__constant` is surrounded by square brackets,
+   the implicit arguments of :n:`@qualid__constant` are ignored
+   when translating to :n:`@qualid__constructor` (i.e., before
+   applying :n:`@qualid__print`) and filled with implicit
+   argument holes :g:`_` when translating from
+   :n:`@qualid__constructor` to :n:`@qualid__constant` (after
+   :n:`@qualid__parse`). See below for an example.
```

Comment:
Best practice is to provide a hyperlink (maybe on "example") to the example since it's more than a screenful away.

Reply:
Thanks, done